### PR TITLE
ForceRegressionTask Update for Frame Averaging Compatibility

### PR DESCRIPTION
**Pull request by melo-gonzo**
The current `ForceRegressionTask` assumes that the `pos` variable will be uses for computing gradients. In Frame Averaging, a transformed version of the position coordinates are used and these updated coordinates should also be used for computing the forces. This PR introduces some logic to handle this for the `stochastic` and `all` Frame Averaging cases (others may work, but not tested).

### CODE REVIEW

File: examples/model_demos/faenet_pyg.py [hunk 1]
```diff
@@ -18,22 +16,22 @@
 """
 
 # construct IS2RE relaxed energy regression with PyG implementation of FAENet
```
Comment (laserkelvin) [inline examples/model_demos/faenet_pyg.py:18]: Update comment since it's S2EF

File: examples/model_demos/faenet_pyg.py [hunk 2]
```diff
@@ -48,43 +46,5 @@
...
-            ),
-            FrameAveraging(frame_averaging="3D", fa_method="stochastic"),
-        ],
-    },
-)
-
-# run a quick training loop
-trainer = pl.Trainer(fast_dev_run=10)
-trainer.fit(task, datamodule=dm)
+trainer = pl.Trainer(enable_checkpointing=False, logger=False, devices=1)
```
Comment (laserkelvin) [inline examples/model_demos/faenet_pyg.py:49]: Is there a reason you're not using `fast_dev_run`?

File: matsciml/models/base.py [hunk 3]
```diff
@@ -1564,53 +1568,71 @@ def forward(
...
-            -1
-            * torch.autograd.grad(
-                energy,
-                pos,
-                grad_outputs=torch.ones_like(energy),
-                create_graph=True,
-            )[0]
-        )
+
+        def energy_and_force(pos, system_embedding):
```
Comment (laserkelvin) [inline matsciml/models/base.py:1593]: I think it's clear from the variable names, but can you add type annotations anyway?

**Review by laserkelvin — CHANGES_REQUESTED** [review]

**Comment by laserkelvin** [comment]
Model pytests failing are unrelated to this PR, and the `pre-commit` hooks that fail are just stylistic ones addressed in #191

File: examples/model_demos/faenet_pyg.py (hunk 1 above)
Comment (melo-gonzo) [reply examples/model_demos/faenet_pyg.py]: updated in [502337c](https://github.com/IntelLabs/matsciml/pull/194/commits/502337cacf13a7e2a83aa7297ebfca7a6a3a1cbe)

File: examples/model_demos/faenet_pyg.py (hunk 2 above)
Comment (melo-gonzo) [reply examples/model_demos/faenet_pyg.py]: updated in [502337c](https://github.com/IntelLabs/matsciml/pull/194/commits/502337cacf13a7e2a83aa7297ebfca7a6a3a1cbe)

**Review by laserkelvin — APPROVED** [review]

File: matsciml/models/base.py (hunk 3 above)
Comment (melo-gonzo) [reply matsciml/models/base.py]: updated in [502337c](https://github.com/IntelLabs/matsciml/pull/194/commits/502337cacf13a7e2a83aa7297ebfca7a6a3a1cbe)
